### PR TITLE
feat: add ability to specify nginx_proxy.conf file

### PR DESCRIPTION
**Pull request by YSSofer**
If there's an `nginx_proxy.conf` file in the user's `$HOME/.docker/` directory.

This will allow for custom nginx configs like setting headers or mutual TLS using x.509 certificates in local setups.

_Note: Also updated gh actions to fix errors_

### CODE REVIEW

**Comment by YSSofer** [comment]
Closed. Duplicated by #39

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "commander": "^2.20.0",
     "dockerode": "^2.5.8",
     "execa": "^1.0.0",
+    "fs": "0.0.1-security",
```
Comment (codfish) [inline package.json]: what's this guy? I would use builtin node `fs`, no dependency needed.

File: src/containers/proxyConfig.js [hunk 2]
```diff
@@ -14,6 +15,14 @@ export default {
     Binds: [
       '/var/run/docker.sock:/tmp/docker.sock:ro',
       `${resolve(homedir(), '.dotdocker/certs')}:/etc/nginx/certs:ro`,
+      ...(fs.existsSync(`${resolve(homedir(), '.dotdocker/nginx_proxy.conf')}`)
```
Comment (codfish) [inline src/containers/proxyConfig.js]: Just a suggestion: pulling this logic up just to make this a little easier to read, something like:

```js
const nginxProxyPath = resolve(homedir(), '.dotdocker/nginx_proxy.conf');
const nginxProxyBindings = fs.existsSync(nginxProxyPath)
  ? [`${nginxProxyPath}:/etc/nginx/conf.d/nginx_proxy.conf:ro`]
  : [];

  // ...

  HostConfig: {
     Binds: [
       '/var/run/docker.sock:/tmp/docker.sock:ro',
       `${resolve(homedir(), '.dotdocker/certs')}:/etc/nginx/certs:ro`,
       ...nginxProxyBindings,
     ]
  }
```

that's assuming @aj-may agrees with this approach?

**Review by codfish — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (YSSofer) [reply package.json]: I thought was required but you right!

File: src/containers/proxyConfig.js (hunk 2 above)
Comment (YSSofer) [reply src/containers/proxyConfig.js]: I like it. @aj-may can revert if you prefer.

**Review by codfish — APPROVED** [review]

**Review by magoon — APPROVED** [review]
Approved - I have been running this for weeks and it works

**Review by aj-may — APPROVED** [review]

**Pull request merged**